android-components/components/browser/state/src/test/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddlewareTest.kt

```diff
@@ -559,4 +560,150 @@ class TranslationsMiddlewareTest {
+                languageSettings = mockLanguageSetting,
+            ),
+        )
+        waitForIdle()
+    }
+
+    @Test
+    fun `WHEN an Operation to FETCH_AUTOMATIC_LANGUAGE_SETTINGS is dispatched THEN SetLanguageSettingsAction is dispatched`() = runTest {
+        // Send Action
+        val action =
+            TranslationsAction.OperationRequestedAction(
+                tabId = tab.id,
+                operation = TranslationOperation.FETCH_AUTOMATIC_LANGUAGE_SETTINGS,
+            )
+        translationsMiddleware.invoke(context = context, next = {}, action = action)
+        waitForIdle()
+
+        // Check expectations
+        val languageSettingsCallback = argumentCaptor<((Map<String, LanguageSetting>) -> Unit)>()
+        // Checking atLeastOnce, because InitAction is also implicitly called earlier
+        verify(engine, atLeastOnce()).getLanguageSettings(
+            onSuccess = languageSettingsCallback.capture(),
+            onError = any(),
+        )
+        val mockLanguageSetting = mapOf("en" to LanguageSetting.OFFER)
+        languageSettingsCallback.value.invoke(mockLanguageSetting)
+        waitForIdle()
+
+        verify(store, atLeastOnce()).dispatch(
+            TranslationsAction.SetLanguageSettingsAction(
+                languageSettings = mockLanguageSetting,
+            ),
+        )
+        waitForIdle()
+    }
+
+    @Test
+    fun `WHEN an Operation to UpdatePageSettings for UPDATE_ALWAYS_TRANSLATE_LANGUAGE is dispatched THEN SetLanguageSettingsAction is dispatched`() = runTest {
+        // Page settings needs additional setup
+        setupMockState()
+        val pageSettingCallback = argumentCaptor<(() -> Unit)>()
+        whenever(
+            engine.setLanguageSetting(
+                languageCode = any(),
+                languageSetting = any(),
+                onSuccess = pageSettingCallback.capture(),
+                onError = any(),
+            ),
+        ).thenAnswer { pageSettingCallback.value.invoke() }
+
+        // Send Action
+        val action =
+            TranslationsAction.UpdatePageSettingAction(
+                tabId = tab.id,
+                operation = TranslationPageSettingOperation.UPDATE_ALWAYS_TRANSLATE_LANGUAGE,
+                setting = true,
+            )
+        translationsMiddleware.invoke(context = context, next = {}, action = action)
+        waitForIdle()
+
+        // Check expectations
+        val languageSettingsCallback = argumentCaptor<((Map<String, LanguageSetting>) -> Unit)>()
+        // Checking twice, because InitAction is also implicitly called earlier
+        verify(engine, atLeast(2)).getLanguageSettings(
+            onSuccess = languageSettingsCallback.capture(),
+            onError = any(),
+        )
+    }
+
+    @Test
+    fun `WHEN an Operation to FETCH_AUTOMATIC_LANGUAGE_SETTINGS has an issue THEN TranslateExceptionAction is dispatched for language setting`() = runTest() {
```

Comment:
```suggestion
    fun `WHEN an Operation to FETCH_AUTOMATIC_LANGUAGE_SETTINGS has an error THEN TranslateExceptionAction is dispatched for language setting`() = runTest() {
```
This might provide better clarity.

Reply:
Updated, good idea!